Ci types format

naddeoa opened the pull request.

Commit:
Add makefile targets and linter/formatter

Commit:
Add CI files

jamie256 commented on pyproject.toml:
```diff
@@ -17,7 +17,37 @@ black = "^23.11.0"
+reportMissingTypeArgumet = false
+
+[tool.ruff]
+line-length = 140
```
we use 160 in whylogs and LangKit, why 140 here?

naddeoa replied:
I use 140 in the integrations, I just copied this over from there. It can be 160.

jamie256 commented on .github/actions/python-build/action.yml:
```diff
@@ -0,0 +1,45 @@
+      with:
+        python-version: ${{ inputs.python_version }}
+
+    - uses: Gr1N/setup-poetry@v8
```
Any reasons for this poetry setup action vs the one in our other repos or something like https://github.com/snok/install-poetry? We should turn on caching (fine to do in follow up).

naddeoa replied:
No reason at all, I just happened to find this one first and I've been using it. Looks like they take the same options more or less so I can sub in the other one.

jamie256 commented on pyproject.toml:
```diff
@@ -17,7 +17,37 @@ black = "^23.11.0"
 build-backend = "poetry.core.masonry.api"
+
+[tool.pyright]
+include = ["whylogs_container", "integ", "tests"]
```
whylogs_container?

naddeoa replied:
oops, will fix.

jamie256 commented on pyproject.toml:
```diff
@@ -17,7 +17,37 @@ black = "^23.11.0"
+[tool.ruff]
+line-length = 140
+indent-width = 4
+include = ["./whylogs_container/**/*.py", "./tests/**/*.py", "./integ/**/*.py"]
```
?

naddeoa replied:
This is how ruff is configured. Ruff is just a nicer, more modern version of black that lets you get rid of all of the various specialized tools for formatting. Don't need to also pull in isort, for example. Pretty fast too and it has an LSP integration

jamie256 replied:
the `whylogs_container` folders were what I was questioning there, you fixed it now.

jamie256 reviewed (commented):
Left some minor comments and questions

naddeoa commented:
@jamie256 Actually there are type errors now that the dir name is fixed. Not sure how to correct them either because I'm not sure what the intent was in the top level __init__.py file. 

![image](https://github.com/whylabs/llm-traceguard/assets/1233709/0f0209c6-50ce-4952-be73-2e83b3e420d3)

Wouldn't this be better achieved with a bump config rather than a runtime dependency?

naddeoa commented:
There are actually a fair bit of errors and I won't be able to fix them. Lots of missing types.

jamie256 commented:
> @jamie256 Actually there are type errors now that the dir name is fixed. Not sure how to correct them either because I'm not sure what the intent was in the top level **init**.py file.
> 
> Wouldn't this be better achieved with a bump config rather than a runtime dependency?

We can use a string literal + bumpconfig to update it, the intention is to have a standard `__version__` available. e.g. https://github.com/protocolbuffers/protobuf/blob/ccfab6a8d552d40cbc78ea9c3de3d00597bb8fc7/python/google/protobuf/__init__.py#L10

Commit:
Add bump config support

This implements the __version__ with a static string version that we
increment with bump config.

Commit:
Change version format from a.b.c.devn a.b.c-devn (dash at end)

For whatever reason, I can't get bump config to work with a fourth
dot section.

naddeoa commented:
> > @jamie256 Actually there are type errors now that the dir name is fixed. Not sure how to correct them either because I'm not sure what the intent was in the top level **init**.py file.
> > Wouldn't this be better achieved with a bump config rather than a runtime dependency?
> 
> We can use a string literal + bumpconfig to update it, the intention is to have a standard `__version__` available. e.g. https://github.com/protocolbuffers/protobuf/blob/ccfab6a8d552d40cbc78ea9c3de3d00597bb8fc7/python/google/protobuf/__init__.py#L10

Updated it to remove that function and added a bump config + static version string in its place
